acceptance: split allocator test outputs

BramGruneir opened the pull request:
This change will split the current output to show both how long the rebalancing
took (to reach a stddev of replica counts of less than 1) and how long it takes
to then reach equilibrium (no changeReplicas for 3 mins).

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7817)

<!-- Reviewable:end -->

BramGruneir committed:
acceptance: split allocator test outputs

This change will split the current output to show both how long the rebalancing
took (to reach a stddev of replica counts of less than 1) and how long it takes
to then reach equilibrium (no changeReplicas for 3 mins).

cuongdo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> w/ comments

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[acceptance/allocator_test.go, line 221 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7817#-KMzkGcE5rD4BL4Y1Uys:-KMzkGcFX7NOhgniXhuC:-573510713) ([raw file](https://github.com/cockroachdb/cockroach/blob/27df5f7337f9234ddf7eb9a1afa9e39383a4db8a/acceptance/allocator_test.go#L221)):*
> ```Go
> // rebalancedAt parameter represents how long it took for the cluster to be
> // rebalanced but not necessarily settle into an equilibrium.
> func (at *allocatorTest) printRebalanceStats(db *gosql.DB, host string, adminPort int, rebalancedAt time.Duration) error {
> ```

`rebalancedAt` implies a `time.Time`. Maybe `rebalanceDuration`?

---

*[acceptance/allocator_test.go, line 348 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7817#-KMziOBeudSRoyrLtaiB:-KMziOBeudSRoyrLtaiC:1341222654) ([raw file](https://github.com/cockroachdb/cockroach/blob/27df5f7337f9234ddf7eb9a1afa9e39383a4db8a/acceptance/allocator_test.go#L348)):*
> ```Go
> 					return err
> 				}
> 				if stddev < 1 {
> ```

When the balancing operation doesn't actually converge, this change will make allocator tests more susceptible to leaking a test cluster. See #7712. Nothing actionable, so just noting this.

---

*[acceptance/allocator_test.go, line 349 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7817#-KMzjmvk5AldxY1dS1Cs:-KMzjmvk5AldxY1dS1Ct:687426316) ([raw file](https://github.com/cockroachdb/cockroach/blob/27df5f7337f9234ddf7eb9a1afa9e39383a4db8a/acceptance/allocator_test.go#L349)):*
> ```Go
> 				}
> 				if stddev < 1 {
> 					rebalancedAt = time.Now().Sub(startTime)
> ```

`rebalancedAt` is an odd name for a duration. Maybe `rebalanceDuration`or `rebalancedAfter`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7817#-:-KMzl1Emoe2r-x_jFWpL:-1477467219)*
<!-- Sent from Reviewable.io -->

BramGruneir committed:
acceptance: add a context to allocatorTest

BramGruneir commented:
Addressed comments and added some context plumbing while I was there.  PTAL (a super quick one)

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[acceptance/allocator_test.go, line 221 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7817#-KMzkGcE5rD4BL4Y1Uys:-KNYAgnF4JKwZIScD4cE:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/27df5f7337f9234ddf7eb9a1afa9e39383a4db8a/acceptance/allocator_test.go#L221)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> `rebalancedAt` implies a `time.Time`. Maybe `rebalanceDuration`?
</details>
Done.

---

*[acceptance/allocator_test.go, line 349 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7817#-KMzjmvk5AldxY1dS1Cs:-KNYAlDW0DKjkkUVghbY:-1436255790) ([raw file](https://github.com/cockroachdb/cockroach/blob/27df5f7337f9234ddf7eb9a1afa9e39383a4db8a/acceptance/allocator_test.go#L349)):*
<details><summary><i>Previously, cuongdo (Cuong Do) wrote…</i></summary>
> `rebalancedAt` is an odd name for a duration. Maybe `rebalanceDuration`or `rebalancedAfter`?
</details>
rebalanceDuration . Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7817#-:-KNYgrYU9xnz4Y-bADiR:752176618)*
<!-- Sent from Reviewable.io -->
